<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>651</width>
    <height>422</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>631</width>
      <height>260</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="PositionTab">
     <attribute name="title">
      <string>Position Controller</string>
     </attribute>
     <widget class="QSlider" name="sldPosition">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>185</y>
        <width>561</width>
        <height>29</height>
       </rect>
      </property>
      <property name="minimum">
       <number>-10000</number>
      </property>
      <property name="maximum">
       <number>10000</number>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="invertedAppearance">
       <bool>false</bool>
      </property>
      <property name="tickPosition">
       <enum>QSlider::TicksAbove</enum>
      </property>
      <property name="tickInterval">
       <number>1000</number>
      </property>
     </widget>
     <widget class="QLabel" name="lblLowerLimit">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>160</y>
        <width>60</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>-1000</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lblUpperLimit">
      <property name="geometry">
       <rect>
        <x>550</x>
        <y>160</y>
        <width>60</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>1000</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>601</width>
        <height>91</height>
       </rect>
      </property>
      <property name="title">
       <string>PID</string>
      </property>
      <widget class="QPushButton" name="btnSetPID">
       <property name="geometry">
        <rect>
         <x>510</x>
         <y>50</y>
         <width>80</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>Set PID</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="editD">
       <property name="geometry">
        <rect>
         <x>510</x>
         <y>20</y>
         <width>80</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>1</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="lblPosCurP">
       <property name="geometry">
        <rect>
         <x>119</x>
         <y>50</y>
         <width>50</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="lblCurD">
       <property name="geometry">
        <rect>
         <x>330</x>
         <y>50</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Current D:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="lblI">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>20</y>
         <width>100</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>Integral(I)</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLineEdit" name="editI">
       <property name="geometry">
        <rect>
         <x>310</x>
         <y>20</y>
         <width>80</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>1</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="lblCurI">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>50</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Current I:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="lblP">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>110</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>Proportion(P)</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="lblPosCurI">
       <property name="geometry">
        <rect>
         <x>260</x>
         <y>50</y>
         <width>50</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLineEdit" name="editP">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>20</y>
         <width>80</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>1</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="lblD">
       <property name="geometry">
        <rect>
         <x>390</x>
         <y>20</y>
         <width>121</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>Derivative(D)</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="lblPosCurD">
       <property name="geometry">
        <rect>
         <x>420</x>
         <y>50</y>
         <width>50</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="lblCurP">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>50</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Current P:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>100</y>
        <width>601</width>
        <height>61</height>
       </rect>
      </property>
      <property name="title">
       <string>Position</string>
      </property>
      <widget class="QLabel" name="lblCurTarget">
       <property name="geometry">
        <rect>
         <x>380</x>
         <y>20</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="lblCurPos">
       <property name="geometry">
        <rect>
         <x>240</x>
         <y>20</y>
         <width>131</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Current Target:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QPushButton" name="btnSetPos">
       <property name="geometry">
        <rect>
         <x>490</x>
         <y>20</y>
         <width>101</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>Set &amp;&amp; Move</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="editPos">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>20</y>
         <width>100</width>
         <height>30</height>
        </rect>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="lblPosition">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>121</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>Target Position</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </widget>
     <widget class="QLabel" name="lblOrigin">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>160</y>
        <width>60</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="SpeedTab">
     <attribute name="title">
      <string>Speed Controller</string>
     </attribute>
    </widget>
   </widget>
   <widget class="QLabel" name="lblTitle">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>-10</y>
      <width>250</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>PID Emulator</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="lblAuthor">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>320</y>
      <width>191</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>(c)2016 Xia Weiwen</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="btnQuit">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>320</y>
      <width>99</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>&amp;Quit</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblWarning">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>290</y>
      <width>451</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
     </font>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>651</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_Quit"/>
   </widget>
   <widget class="QMenu" name="menuAbout_2">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="actionAbout_Qt"/>
   </widget>
   <addaction name="menuAbout"/>
   <addaction name="menuAbout_2"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action_Quit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="text">
    <string>About Qt</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
